src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java

```diff
@@ -287,6 +295,66 @@ public Response uploadIrbDocument(
     }
   }
 
+  @POST
+  @Consumes(MediaType.MULTIPART_FORM_DATA)
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/progress_report/{parentReferenceId}")
+  @RolesAllowed({RESEARCHER})
+  public Response postProgressReport(
+      @Auth AuthUser authUser,
+      @PathParam("parentReferenceId") String parentReferenceId,
+      @FormDataParam("dar") String dar,
+      @FormDataParam("collaboratorRequiredFile") InputStream collabInputStream,
+      @FormDataParam("collaboratorRequiredFile") FormDataContentDisposition collabFileDetails,
+      @FormDataParam("ethicsApprovalRequiredFile") InputStream ethicsInputStream,
+      @FormDataParam("ethicsApprovalRequiredFile") FormDataContentDisposition ethicsFileDetails) {
+    User user = userService.findUserByEmail(authUser.getEmail());
+    DataAccessRequest parentDar = dataAccessRequestService.findByReferenceId(parentReferenceId);
+    checkAuthorizedUpdateUser(user, parentDar);
+    DataAccessRequest payload = populateDarFromJsonString(user, dar);
+    DataAccessRequest childDar = dataAccessRequestService.createDataAccessRequest(user, payload);
+
+    for (Integer datasetId : childDar.getDatasetIds()) {
```

Comment:
There's an issue here if the user passes in a json body with no dataset ids. I think the bug is in `populateDarFromJsonString`. Passing in `{"projectTitle": "PR Testing"}` yields this error:
```
{
  "code": 404,
  "message": "Dataset 0 not found"
}
```

Reply:
Bug fixed in latest commit.